examples/wordCount/lines2words/lines2words.cpp

```diff
@@ -99,52 +59,65 @@ std::vector<uint32_t> createConsumerPartitions(iceflow::Edge upstreamEdge,
   return consumerPartitions;
 }
 
-int main(int argc, const char *argv[]) {
-
-  if (argc != 3) {
-    std::string command = argv[0];
-    std::cout << "usage: " << command
-              << " <application-dag-file> <instance-number>" << std::endl;
-    return 1;
-  }
+void run(const std::string &nodeName, const std::string &dagFileName,
+         uint32_t consumerIndex) {
+  WordSplitter wordSplitter;
+  ndn::Face face;
 
-  std::string nodeName = "lines2words";
-  std::string dagFileName = argv[1];
-  int consumerIndex = std::stoi(argv[2]);
   auto dagParser = iceflow::DAGParser::parseFromFile(dagFileName);
-
-  std::string syncPrefix = "/" + dagParser.getApplicationName();
-  auto nodePrefix = generateNodePrefix();
   auto node = dagParser.findNodeByName(nodeName);
-  auto upstreamEdges = dagParser.findUpstreamEdges(node);
 
-  // TODO: Do this dynamically for all edges
-  auto downstreamEdge = node.downstream.at(0);
-  auto downstreamEdgeName = downstreamEdge.id;
-  auto numberOfProducerPartitions = downstreamEdge.maxPartitions;
+  auto upstreamEdge = dagParser.findUpstreamEdges(node).at(0).second;
+  auto upstreamEdgeName = upstreamEdge.id;
+  auto downstreamEdgeName = node.downstream.at(0).id;
```

Comment:
I was wondering if these should be hard-coded in the example or passed via the CLI instead of extracting these from the DAG. That would probably make things a bit more concise.

Reply:
I guess this depends on whether we want people to use the DAG as the primary method of configuring IceFlow.

Reply:
Hmm, yeah, ideally, the DAG is what should be used here. When it comes to a more concise API, maybe we could also extend the DAG parser with methods to find an edge name by node and index. That might make the examples a bit cleaner already.

Reply:
I think I would probably add that functionality via a follow-up PR at this point, if that is okay for you.

Reply:
Just opened https://github.com/hsel-netsys/iceflow/issues/117 as a tracking issue, will resolve this thread now :)